### PR TITLE
IExplosion and stuff

### PR DESCRIPTION
**Pull request by democat3457**
The time has come, CTUtils' IExplosion has now been ported to the main mod. BrisingrAerowing/CraftTweakerUtils#7

It also fixes build errors that I accidentally introduced, as well as fixes the JavaDoc warnings and makes the gitpod configuration actually work.

Documentation coming soon:tm:.

### CODE REVIEW

File: CraftTweaker2-API/src/main/java/crafttweaker/api/world/IWorld.java [hunk 1]
```diff
@@ -104,7 +104,25 @@ default IRayTraceResult rayTraceBlocks(IVector3d begin, IVector3d ray, @Optional
 
     @ZenMethod
     default IItemStack getPickedBlock(IBlockPos pos, IRayTraceResult rayTraceResult, IPlayer player) {
-        CraftTweakerAPI.logError(this.getClass().getName() + " does not override IWorld.getPickedBlock!");
+        CraftTweakerAPI.logError(this.getClass().getName() + " does not override IWorld.getPickedBlock, tell the author to fix that.");
+        return null;
+    }
+
+    @ZenMethod
+    default IExplosion createExplosion(IEntity exploder, double x, double y, double z, float strength, boolean causesFire, boolean damagesTerrain) {
```
Comment (friendlyhj) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/world/IWorld.java:112]: As you know, the `exploder` parameter in Java method is Nullable, but according to the my previous working, the parameter cannot be null in zenscript. And the `@Optional` annotation only can be used to the last parameter. Maybe you can overload the method , which omits the `exploder` parameter?

Comment (democat3457) [reply CraftTweaker2-API/src/main/java/crafttweaker/api/world/IWorld.java:112]: Would changing IEntity to IAny and then checking if the object is an instanceof IEntity work instead?

Comment (friendlyhj) [reply CraftTweaker2-API/src/main/java/crafttweaker/api/world/IWorld.java:112]: IAny? I wonder whether it accepts null or not. I mean overloading the method. Add
```java
 @ZenMethod
    default IExplosion createExplosion(double x, double y, double z, float strength, boolean causesFire, boolean damagesTerrain) {
    this.createExplosion(null, x, y, z, strength, causesFire, damagesTerrain);
}
```

Comment (democat3457) [reply CraftTweaker2-API/src/main/java/crafttweaker/api/world/IWorld.java:112]: Yeah I know what overloading is, but it's more convenient to just be able to use null at the beginning to explicitly say no entity

Comment (democat3457) [reply CraftTweaker2-API/src/main/java/crafttweaker/api/world/IWorld.java:112]: image

apparently null works with non-primitives

**Comment by democat3457** [comment]
"ready for review" @jaredlll08

**Comment by jaredlll08** [comment]
Looks good to me, just resolve the crafttweaker_at.cfg and I'll merge

**Comment by democat3457** [comment]
hath been resolved

**Pull request merged**